[David] iP

vvidday opened the pull request:
# Ekud
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes))

Ekud frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://github.com/vvidday/ip/releases/).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you're a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

vvidday committed:
changed name of chatbot

vvidday committed:
bot echoes user commands and exits upon bye

vvidday committed:
refactor: refactored ekud code into class

vvidday committed:
feat: chat bot stores text from user and displays data when requested

vvidday committed:
refactor: refactored switch case to if else to support string operations

vvidday committed:
feat: ability to mark tasks as done

vvidday committed:
feat: ability to mark tasks as undone

vvidday committed:
refactor: refactored to use Task toString() method

vvidday committed:
feat: subtypes of task class (todo, event, deadline)

vvidday committed:
feat: support for todos, deadlines and events

vvidday committed:
test: reconfigured test files to follow new naming convention, added basic test case

vvidday committed:
refactor: refactored errors into custom exception

vvidday committed:
refactor: refactored task array into arraylist

vvidday committed:
feat: support for deleting tasks from list

vvidday committed:
test: updated test cases to test delete feature

vvidday committed:
docs: javadocs for public facing methods

vvidday committed:
fix: properly isolate first word in command to make parsing more accurate

vvidday committed:
Add storage class and methods to handle creating directory/txt file

vvidday committed:
Add method to tasks that converts details to string to be stored in file

vvidday committed:
Add method to save to file after every action by user

vvidday committed:
Add data folder to gitignore

vvidday committed:
Add functionality to read and instantiate tasks from txt file

vvidday committed:
Add date support for deadline

vvidday committed:
Merge branch 'branch-Level-7'

vvidday committed:
Merge branch 'branch-Level-8'

vvidday committed:
Change deadline parsing to work with actual date instead of string

vvidday committed:
Refactor logic into separate classes

vvidday committed:
Refactor classes into packages

vvidday committed:
Merge branch 'add-gradle-support' of github.com:nus-cs2103-AY2223S1/ip

vvidday committed:
Change build.gradle to include checkstyle and enable assertions

vvidday committed:
Add basic tests for Task class and Todo/Event/Deadline

vvidday committed:
Change mainclassname and archivebasename of build.gradle for jar files

vvidday committed:
Add javadocs for task package

vvidday committed:
Add javadocs for util package

vvidday committed:
Add javadocs for Deadline.java::getDateFromString method

vvidday committed:
Add private method to print tasks of a custom list and refactor current public method

vvidday committed:
Add find functionality

vvidday committed:
Merge branch 'branch-A-JavaDoc'

vvidday committed:
Merge branch 'branch-Level-9'

vvidday committed:
Fix formatter to indent 4 spaces instead of 2

vvidday committed:
Fix misc issues relating to coding standard & code quality

vvidday committed:
Remove checkstyle from gradle temporarily

vvidday committed:
Add fxml and image files to resources

vvidday committed:
Add gui controllers for dialogbox and mainwindow

vvidday committed:
Modify ekud to work with new gui instead of scanner input

vvidday committed:
Add launcher and main class to start application with gui

vvidday committed:
Merge branch 'branch-Level-10'

jamietan2002 commented on src/main/java/ekud/main/Ekud.java:
```diff
@@ -0,0 +1,49 @@
+        }
+    }
+
+}
```
neat code, adheres to coding standards

jamietan2002 commented on src/main/java/ekud/task/Deadline.java:
```diff
@@ -0,0 +1,58 @@
+     * @param dueDateString Due Date of task in YYYY-MM-DD format.
+     * @throws EkudException Exception in creation of deadline due to invalid due
+     *         date.
+     */
```
javadocs is good

jamietan2002 commented on src/main/java/ekud/task/Deadline.java:
```diff
@@ -0,0 +1,58 @@
+     * @return LocalDate object representing date.
+     * @throws EkudException Error that occurred, due to invalid string format.
+     */
+    private LocalDate getDateFromString(String dateString) throws EkudException {
```
method name is clear

jamietan2002 commented on src/main/java/ekud/task/Event.java:
```diff
@@ -0,0 +1,33 @@
+    public String toString() {
+        return String.format("[E]%s (at: %s)", super.toString(), this.at);
+    }
+}
```
neat code

jamietan2002 commented on src/main/java/ekud/task/Task.java:
```diff
@@ -0,0 +1,76 @@
+    /**
+     * Returns string representing details of Task to be stored in the txt file.
+     * 
+     * @return String representing details of Task
```
do you need to add a fullstop?

jamietan2002 commented on src/main/java/ekud/task/TaskList.java:
```diff
@@ -0,0 +1,210 @@
+        return (String.format("Got it. I've added this task: %s", task.toString()));
+    }
+
+    private Task parseSpecialTask(String description, TaskType type) throws EkudException {
```
what is special task?

jamietan2002 reviewed (commented):
Overall adheres to coding standards good job!

MuhdMusab commented on src/main/java/ekud/main/Ekud.java:
```diff
@@ -0,0 +1,49 @@
+        this.storage = new Storage();
+        this.ui = new Ui();
+        this.parser = new Parser();
+        TaskList taskList = null;
```
Is this initialization necessary?

MuhdMusab commented on src/main/java/ekud/task/TaskList.java:
```diff
@@ -0,0 +1,210 @@
+     */
+    public String deleteTask(String message) throws EkudException {
+        if (message.matches("^delete \\d+$")) {
+            int idx = Integer.parseInt(message.substring("delete ".length()));
```
Maybe use a more descriptive variable name here?

MuhdMusab commented on src/main/java/ekud/task/TaskList.java:
```diff
@@ -0,0 +1,210 @@
+
+    private Task parseSpecialTask(String description, TaskType type) throws EkudException {
+        Optional<Task> task = Optional.empty();
+        String[] parts = description.split(" ");
```
Similarly, maybe a more descriptive variable name could be used here.

MuhdMusab commented on src/main/java/ekud/util/Parser.java:
```diff
@@ -0,0 +1,54 @@
+        }
+    }
+
+}
```
Goodjob 👍. Very neat and easy to read.

MuhdMusab reviewed (commented):
Good job. Good use of guard clauses and coding principles. A few cosmetic changes suggested but well done.

vvidday committed:
Add assertions to task classes

The current code only checks if getStatusIcon() returns 'X' or not,
but the only two possible status icons should be 'X' or ' '.

Let's,
* update todo, deadline and event class to assert if getStatusIcon() returns either
of the two valid status icons.

vvidday committed:
Add assertion to check if task is defineed when decoding

The tasks in the .txt file should start with either T,E or D.
There might be a scenario where the .txt file is modified by
the user, resulting in this to not be the case.

Let's write an assertion to assert that the task is not null
at that point.

vvidday committed:
Use .equals() instead of == for comparing strings

There are three instances where the code uses == to compare strings
instead of .equals(), which is the proper way of comparing strings in
java.

Let's change those to use .equals() instead.

vvidday committed:
Add default branch to switch statement in storage

vvidday committed:
Merge pull request #2 from vvidday/branch-A-Assertions

Branch A-Assertions

vvidday committed:
Resolve merge conflicts

vvidday committed:
Merge pull request #1 from vvidday/branch-A-CodeQuality

Branch A-CodeQuality

vvidday committed:
Add github actions for CI

vvidday committed:
Add note and noteList class

vvidday committed:
Modify existing classes to work with new note functionality

vvidday committed:
Change images for user and bot

vvidday committed:
Update README.md and add UI.png

vvidday committed:
Update README.md
